src/compiler/checker.ts

```diff
@@ -4259,7 +4259,14 @@ namespace ts {
                 }
 
                 if (isPropertyAccessExpression(expression.left) && expression.left.expression.kind === SyntaxKind.ThisKeyword) {
-                    if (getThisContainer(expression, /*includeArrowFunctions*/ false).kind === SyntaxKind.Constructor) {
+                    const thisContainer = getThisContainer(expression, /*includeArrowFunctions*/ false);
+                    const isPrototypeProperty = isBinaryExpression(thisContainer.parent) &&

```

Comment:
Only used if `thisContainer.kind === SyntaxKind.FunctionExpression`, might want to avoid computing this otherwise.